src/meta/meta_ckpt.c

```diff
@@ -1262,6 +1262,152 @@ __wt_meta_ckptlist_update_config(
+    WT_DECL_RET;
+    char *file_config;
+
+    WT_CLEAR(blocks);
+    WT_CLEAR(key);
+    WT_CLEAR(backup_config_value);
+    WT_CLEAR(blocks_key);
+    WT_CLEAR(blocks_value);
+    WT_CLEAR(*output_item);
+
+    metadata_cursor = NULL;
+    file_config = NULL;
+
+    WT_RET(__wt_metadata_cursor_open(session, NULL, &metadata_cursor));
+
+    metadata_cursor->set_key(metadata_cursor, uri);
+    WT_ERR(metadata_cursor->search(metadata_cursor));
+    WT_ERR(metadata_cursor->get_value(metadata_cursor, &file_config));
+    WT_ERR(
+      __wt_config_getones(session, file_config, "checkpoint_backup_info", &backup_config_value));
+
+    if ((backup_config_value.len > 0) && (backup_config_value.type == WT_CONFIG_ITEM_STRUCT)) {
+        __wt_config_subinit(session, &blkconf, &backup_config_value);
+
+        /* Loop through the incremental backup blocks data looking for the correct id */
+        while ((ret = __wt_config_next(&blkconf, &blocks_key, &blocks_value)) == 0) {
+            if (blocks_value.len > 0) {
+                /* Have we found the blocks data for the correct key (which has id 'id')? */
+                if (WT_STRING_MATCH(id, blocks_key.str, blocks_key.len) == 0)
+                    continue; /* Keep searching */
+
+                /* We've found the right blocks so read the bit pattern into output_item */
+                ret = __wt_config_subgets(session, &blocks_value, "blocks", &blocks);
+                if ((ret == 0) && (blocks.len > 0)) {
+                    ret = __wt_nhex_to_raw(session, blocks.str, blocks.len, output_item);
+                    break;
+                }
+            }
+        }
+    }
+
+    if (ret == WT_NOTFOUND)
+        ret = 0;
+
+err:
+    WT_RET(metadata_cursor->close(metadata_cursor));
+
+    return (ret);
+}
+
+/*
+ * check_incorrect_modified_bits --
+ *     This function takes as input two bitmaps (in WT_ITEMs), an original and a new. If any bits in
+ *     the original changed from 1 to 0, it's an error, and the new bitmap is NOT ok. Otherwise the
+ *     new bitmap is ok.
+ */
+static int
+check_incorrect_modified_bits(WT_ITEM *original_bitmap, WT_ITEM *new_bitmap, bool *ok)
+{
+    size_t index;
+    unsigned char partial_result;
+    unsigned char *original_ptr, *new_ptr;
+
+    index = 0;
+    partial_result = 0;
+    original_ptr = NULL;
+    new_ptr = NULL;
+    *ok = false;
+
+    if ((original_bitmap == NULL) || (new_bitmap == NULL))
+        WT_RET(EINVAL);
+
+    /* If the new bitmap is smaller, then some bits have been lost which is a problem. */
+    if (original_bitmap->size > new_bitmap->size)
+        WT_RET(EINVAL);
+
+    original_ptr = (unsigned char *)original_bitmap->data;
+    new_ptr = (unsigned char *)new_bitmap->data;
+
+    *ok = true;
+
+    for (index = 0; index < original_bitmap->size; index++) {
+        /* Detect bits that were (incorrectly) changed from 1 in the original to 0 in the new */
+        partial_result = *original_ptr & ~(*new_ptr);
```

Comment:
I'm absolutely all for using tried and true well-tested functions whenever possible. However in this case, I think that the proposed code

```
    for (index = 0; index < original_bitmap->size; index++) {
        /* Detect bits that were (incorrectly) changed from 1 in the original to 0 in the new */
        partial_result = *original_ptr & ~(*new_ptr);

        if (partial_result != 0) {
            *ok = false;
            break;
        }

        original_ptr++;
        new_ptr++;
    }
```

is simpler (and possibly faster) than using __bit_ffc(). What do you think?

Reply:
Yes, I agree that your code is much shorter and simpler. Can you elaborate on the comment about detecting the bits using bitwise complement (so that readers less familiar with C's bitwise operations have a phrase to search).

Reply:
Hi @sueloverso , I've added a comment to describe this in more detail. How does that look now?